rust/noosphere-cli/tests/peer_to_peer.rs

```diff
@@ -227,8 +240,27 @@ async fn synchronize_petnames_as_they_are_added_and_removed() -> Result<()> {
             ctx.set_petname("thirdparty", None).await?;
             ctx.save(None).await?;
             ctx.sync(SyncRecovery::Retry(3)).await?;
+            wait(1).await;
+            ctx.sync(SyncRecovery::Retry(3)).await?;
```

Comment:
These extra syncs are suspicious, what changes are expected to be pushed or pulled from this second sync? These are of course integration tests fighting with real timing/race conditions where waiting will be necessary, but wondering what e.g. fails without these

Reply:
The gateway does not resolve names synchronously. First sync: client pushes name entries to gateway. Second sync: client gets the gateway's resolved link records.

Two syncs simulates the intended flow.

Reply:
Was missing some context here for syncing a *removed* entry, makes sense!